Fix for aggregate queries

Minerva18 opened the pull request:
Fix for aggregate queries where queryXtra:false will prevent adding limit clause to SOQL. This previously led to a malformed query error.

vineet-sinha commented:
@Minerva18 - Thanks for the change suggestions. Can you make a few minor changes:

- Code: You have `if (params.queryXtra == ''`, shouldn't it be `if (params.queryXtra != false`? Otherwise you prevent the `limit 100` guard to be applied for other queries.
- Documentation: For the parameter so that others calling it will know how to use what you did. Preferably use the JSDoc notation, you can see an example here: https://github.com/salesforce/violet-conversations/blob/master/lib/violetStoreSF.js#L166
- Unit tests: To make sure that your use case keeps working in the future. See here for an example: https://github.com/salesforce/violet-conversations/blob/master/test/test_violetStoreSF.js

Minerva18 commented:
@vineet-sinha Sure! Thank you for the inputs.

You are correct regarding the if condition in the code. I have changed it to `if (params.queryXtra !== false`.

I have also added the documentation similar to the example JSDoc notion and written the unit tests.

Also, there was a typo here - https://github.com/salesforce/violet-conversations/blob/master/lib/violetStoreSF.js#L354. I guess it should be delete instead of update, so I have changes that one too.

Please let me know if there is anything else I can help with.

Thanks!
